pkg/build/reaper/reaper.go

```diff
@@ -0,0 +1,59 @@
+package reaper
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	kapi "k8s.io/kubernetes/pkg/api"
+	kerrors "k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/kubectl"
+
+	"github.com/openshift/origin/pkg/build/util"
+	"github.com/openshift/origin/pkg/client"
+)
+
+// NewBuildConfigReaper returns a new reaper for buildConfigs
+func NewBuildConfigReaper(oc *client.Client) kubectl.Reaper {
+	return &BuildConfigReaper{oc: oc, pollInterval: kubectl.Interval, timeout: kubectl.Timeout}
+}
+
+// BuildConfigReaper implements the Reaper interface for buildConfigs
+type BuildConfigReaper struct {
+	oc                    client.Interface
+	pollInterval, timeout time.Duration
+}
+
+// Stop deletes the build configuration and all of the associated builds.
+func (reaper *BuildConfigReaper) Stop(namespace, name string, timeout time.Duration, gracePeriod *kapi.DeleteOptions) (string, error) {
+	// If the config is already deleted, it may still have associated
+	// builds which didn't get cleaned up during prior calls to Stop. If
+	// the config can't be found, still make an attempt to clean up the
+	// builds.
+	//
+	// We delete the config first to ensure no further builds will be created.
```

Comment:
@wanghaoran1988 Yeah, I've tested this case and it does happen as you say. Seems like we won't do without a controller. @kargakis is it permissible for the reaper to leave the BuildConfig as "deletion pending" in case some build fails to get deleted?